### ISSUE TITLE
(--strings-fmf) refutational soundness bug on QF_S formula

### ISSUE THREAD

**Issue by wintered** [CVC4/CVC4#4735]
The trace below shows a refutational soundness bug in cvc4 in QF_S logic.
```
[605] % cvc4-1.8 small.smt2
sat
[606] % cvc4-1.8 --strings-fmf small.smt2
sat
[607] % cvc4 small.smt2
sat
[608] % cvc4 --strings-fmf small.smt2
unsat
[609] % 
[609] % cat small.smt2
(set-logic QF_S)
(declare-fun a () String)
(declare-fun b () Int)
(declare-fun c () String)
(declare-fun d () String)
(declare-fun e () String)
(declare-fun f () String)
(declare-fun g () String)
(declare-fun h () String)
(declare-fun i () Bool)
(declare-fun j () String)
(declare-fun k () String)
(assert (= e "0000000000"))
(assert (distinct a d))
(assert
 (ite (distinct b 0)
 (and (= (str.++ d e) (str.++ c j))
  (= (str.len c) 2)
  (= j (str.++ f k))
  (= f (str.++ g h))
  (str.in_re g (str.to_re "A")))
 (str.in_re j (str.to_re ""))))
(assert (distinct i (= b 0)))
(assert i)
(check-sat)
[610] %
```
OS: Ubuntu 18.04
Commit: 4b86268

**Comment by wintered** [comment]
another simpler instance for the same bug:
```
[625] % cvc4-1.8 small.smt2
sat
[626] % cvc4-1.8 --strings-fmf small.smt2
sat
[627] % cvc4 small.smt2
sat
[628] % cvc4 --strings-fmf small.smt2
unsat
[629] % 
[629] % cat small.smt2
(set-logic QF_S)
(declare-fun a () String)
(declare-fun b () String)
(assert (distinct b (str.++ a a)))
(assert (str.in_re (str.++ b "\x2f" b) (re.++ (re.opt re.allchar) (str.to_re "\x2f\x65") re.all)))
(check-sat)
[630] %
```
OS: Ubuntu 18.04
Commit: 4b86268

**Comment by 4tXJ7f** [comment]
Fixed in #4736.
